pkg/gameservers/controller.go

```diff
@@ -334,8 +334,21 @@ func (ext *Extensions) creationMutationHandlerPod(review admissionv1.AdmissionRe
 
 	ext.baseLogger.WithField("pod.Name", pod.Name).Debug("creationMutationHandlerPod")
 
-	// TODO: We need to deal with case of multiple and mixed type ports before enabling the feature gate.
-	pod.Spec.Containers[1].Ports[0].ContainerPort = pod.Spec.Containers[1].Ports[0].HostPort
+	annotation, ok := pod.ObjectMeta.Annotations[agonesv1.PassthroughPortAssignmentAnnotation]
+	if !ok {
+		ext.baseLogger.WithField("pod.Name", pod.Name).Debug("creationMutationHandlerPod, " + agonesv1.PassthroughPortAssignmentAnnotation + ": " + pod.ObjectMeta.Annotations[agonesv1.PassthroughPortAssignmentAnnotation])
```

Comment:
I don't think you need to log the value here because you know it's empty. Just say that the annotation is empty and it's unexpected. Feel free to bump it to an `Info`, too, since it's unexpected (so potentially more relevant than debug), but not really that important as it's just noise.

Separately I think you still want to return here after logging, no? I presume the unmarshal will fail o/w.